lib/src/worker/worker_connection.dart

```diff
@@ -2,15 +2,17 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'worker_protocol.pb.dart';
+import 'dart:async';
+
+import '../worker_protocol.pb.dart';
 
 /// Interface for a [WorkerConnection].
 ///
 /// Use [SyncWorkerConnection] or [AsyncWorkerConnection] implementations.
 abstract class WorkerConnection {
   /// Read a [WorkRequest]. Returns either [Future<WorkRequest>] or
```

Comment:
Restructured the comments in this file, and also moved all the implementations into this file. Should be more clear now and hopefully there are fewer useless comments.